tools/src/test/java/org/cyclonedx/schema/JsonSchemaVerificationTest.java

```diff
@@ -13,68 +13,130 @@
+import com.networknt.schema.SchemaValidatorsConfig;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
-
-import org.cyclonedx.parsers.JsonParser;
-import org.cyclonedx.Version;
+import org.apache.commons.lang3.StringUtils;
 import org.junit.jupiter.api.DynamicTest;
 import org.junit.jupiter.api.TestFactory;
 
-import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.junit.jupiter.api.Assertions.assertFalse;
+class JsonSchemaVerificationTest extends BaseSchemaVerificationTest {
+
+    private static final ObjectMapper MAPPER = new JsonMapper();
+
+    private static final JsonSchema VERSION_12;
+    private static final JsonSchema VERSION_13;
+    private static final JsonSchema VERSION_14;
+    private static final JsonSchema VERSION_15;
+    private static final JsonSchema VERSION_16;
+
+    static {
+        JsonMetaSchemaFactory metaSchemaFactory = new DefaultJsonMetaSchemaFactory() {
+            @Override
+            public JsonMetaSchema getMetaSchema(
+                    String iri, JsonSchemaFactory schemaFactory, SchemaValidatorsConfig config) {
+                return addCustomKeywords(super.getMetaSchema(iri, schemaFactory, config));
+            }
+        };
+        JsonSchemaFactory factory = JsonSchemaFactory.builder()
+                .defaultMetaSchemaIri(SchemaId.V7)
+                .metaSchema(addCustomKeywords(JsonMetaSchema.getV7()))
+                .metaSchemaFactory(metaSchemaFactory)
+                .build();
+        ClassLoader cl = JsonSchemaVerificationTest.class.getClassLoader();
+        try {
+            VERSION_12 = factory.getSchema(
+                    requireNonNull(cl.getResource("bom-1.2-strict.schema.json")).toURI());
+            VERSION_13 = factory.getSchema(
+                    requireNonNull(cl.getResource("bom-1.3-strict.schema.json")).toURI());
+            VERSION_14 = factory.getSchema(
+                    requireNonNull(cl.getResource("bom-1.4.schema.json")).toURI());
+            VERSION_15 = factory.getSchema(
+                    requireNonNull(cl.getResource("bom-1.5.schema.json")).toURI());
+            VERSION_16 = factory.getSchema(
+                    requireNonNull(cl.getResource("bom-1.6.schema.json")).toURI());
+        } catch (URISyntaxException e) {
+            throw new IllegalStateException(e);
+        }
+    }
 
-public class JsonSchemaVerificationTest extends BaseSchemaVerificationTest {
+    private static JsonMetaSchema addCustomKeywords(JsonMetaSchema metaSchema) {
+        return JsonMetaSchema.builder(metaSchema)
+                // Non-standard keywords in the CycloneDX schema files.
+                .keyword(new NonValidationKeyword("deprecated"))
+                .keyword(new NonValidationKeyword("meta:enum"))
+                .unknownKeywordFactory(new DisallowUnknownKeywordFactory())
```

Comment:
There are some non standard keywords in the CycloneDX JSON schema files, which requires a custom schema factory configuration.

Reply:
which keywords are these?

Reply:
> which keywords are these? besides `deprecated` and `meta:enum`.

I only found those two.